fluent-dom/src/dom_localization.js

```diff
@@ -0,0 +1,306 @@
+import overlayElement from './overlay';
+import Localization from './localization';
+
+/**
+ * The `DOMLocalization` class is responsible for fetching resources and
+ * formatting translations.
+ *
+ * It implements the fallback strategy in case of errors encountered during the
+ * formatting of translations and methods for observing DOM
+ * trees with a `MutationObserver`.
+ */
+export default class DOMLocalization extends Localization {
+  /**
+   * @returns {DocumentLocalization}
+   */
+  constructor(doc, resIds, generateMessages) {
+    super(doc.location.href, resIds, generateMessages);
+    this.document = doc;
+    this.query = '[data-l10n-id]';
+
+    // A Set of DOM trees observed by the `MutationObserver`.
+    this.roots = new Set();
```

Comment:
Should we initialize `this.roots` with the `document` passed to the constructor?